doc/recipe/preprocessor.rst

```diff
@@ -180,22 +180,68 @@ To get an overview on data fixes and how to implement new ones, please go to
 
 Fx variables as cell measures or ancillary variables
 ====================================================
-The following preprocessor may require the use of ``fx_variables`` 
-to be able to perform the computations:
+The following preprocessor may require the use of ``fx_variables`` to be able
+to perform the computations:
 
-    - ``area_statistics``
-    - ``mask_landsea``
-    - ``mask_landseaice``
-    - ``volume_statistics``
-    - ``weighting_landsea_fraction``
+    - area_statistics_ (``areacella``, ``areacello``)
+    - :ref:`mask_landsea<land/sea/ice masking>` (``sftlf``, ``sftof``)
+    - :ref:`mask_landseaice<ice masking>` (``sftgif``)
+    - volume_statistics_ (``volcello``)
+    - :ref:`weighting_landsea_fraction<land/sea fraction weighting>` (``sftlf``, ``sftof``)
+
+If no ``fx_variables`` are specified for these preprocessors, the fx variables
+given in brackets are used. If given, the ``fx_variables`` argument specifies
+the fx variables that the user wishes to input to the corresponding
+preprocessor function; the user may specify it calling the variables e.g.
+
+.. code-block:: yaml
+
+    fx_variables:
+      areacello:
+      volcello:
+
+or calling the variables and adding specific variable parameters (the key-value
+pair may be as specific as a CMOR variable can permit):
+
+.. code-block:: yaml
+
+    fx_variables:
+      areacello:
+        mip: Omon
+      volcello:
+        mip: fx
+
+Alternatively, the ``fx_variables`` argument can also be specified as a list:
+
+.. code-block:: yaml
+
+    fx_variables: ['areacello', 'volcello']
+
+or as a list of dictionaries:
+
+.. code-block:: yaml
+
+    fx_variables: [{'short_name': 'areacello', 'mip': 'Omon'}, {'short_name': 'volcello', 'mip': 'fx'}]
+
+The recipe parser will automatically find the data files that are associated
+with these variables and pass them to the function for loading and processing.
+
+If ``mip`` is not given, ESMValTool will search for the fx variable in all
+available tables of the specified project.
+
+.. note::
+   Some fx variables exist in more than one table (e.g., ``volcello`` exists in
+   CMIP6's ``Ofx``, ``Oyr``, ``Odec`` and ``Omon`` tables or ``sftgif`` exists
+   in CMIP6's ``fx``, ``LImon``, ``IyrGre`` and ``IyrAnt`` tables). In these
+   cases, ``mip`` needs to be specified, otherwise an error is raised.
```

Comment:
Perhaps it's better to say "In case of ambiguity, ..." because sometimes the variable may be in different tables for different datasets, but every dataset individually is unambiguous.

Reply:
No - this is exactly what the code does, see also my comment below.